docs/docs/workflows/terraform-module.md

```diff
@@ -0,0 +1,119 @@
+    name        = "aws-example"
+    zone        = "us-east-2c"
+    name_prefix = "example"
+    node_groups = {
+      control_plane_default = {
+        role          = "control-plane"
+        zone          = "us-east-2c"
+        instance_type = "m6a.xlarge"
+        disk_size     = 30
+        disk_type     = "gp3"
+        initial_count = 3
+      },
+      worker_default = {
+        role          = "worker"
+        zone          = "us-east-2c"
+        instance_type = "m6a.xlarge"
+        disk_size     = 30
+        disk_type     = "gp3"
+        initial_count = 2
+      }
+    }
+  }
+  ```
+
+  </tabItem>
+
+  <tabItem value="gcp" label="GCP">
+
+  ```
+  module "gcp-constellation" {
+    source = "https://github.com/edgelesssys/constellation/releases/download/$VERSION/terraform-module.zip//modules/gcp-constellation" # TODO check the link during the release. So far only tested with https://tfmodule.s3.eu-west-1.amazonaws.com/modulesV2.zip//modules/aws-constellation
+    name    = "gcp-ex"
+    project = "constellation-331613"
+    service_account_id = "ex"
+    image   = "$VERSION" // replace with the desired version, e.g., `v2.13.0`.
+    zone    = "europe-west2-a"
+    node_groups = {
+      control_plane_default = {
+        role          = "control-plane"
+        zone          = "europe-west2-a"
+        instance_type = "n2d-standard-4"
+        disk_size     = 30
+        disk_type     = "pd-ssd"
+        initial_count = 3
+      },
+      worker_default = {
+        role          = "worker"
+        zone          = "europe-west2-a"
+        instance_type = "n2d-standard-4"
+        disk_size     = 30
+        disk_type     = "pd-ssd"
+        initial_count = 2
+      }
+    }
+  }
+  ```
+
+  </tabItem>
+
+  </tabs>
+
+3. Initialize and apply the module.
+  ```bash
+  terraform init
+  terraform apply
+  ```
+
+## Custom setup
+If you need to separate IAM and cluster management or need custom infrastructure, you can also call the submodules individually.
+You can have a look at the respective convenience module (`{csp}-constellation`) for how the module calls can be structured.
+The submodules are:
+- `constellation-cluster`: applies the Constellation cluster
+- `fetch-image`: translates the Constellation image version to the image ID of the cloud service provider
+- `infrastructure/{csp}`: contains the cluster infrastructure resources
+- `infrastructure/iam/{csp}`: contains the IAM resources used within the cluster
+
+## Cluster upgrades
+The usage of a [remote URL as module source](https://developer.hashicorp.com/terraform/language/modules/sources#fetching-archives-over-http) is recommended, because it simplifies the upgrade process. For [local paths as module source](https://developer.hashicorp.com/terraform/language/modules/sources#local-paths), the user needs to manually overwrite the Terraform files in the Terraform workspace. The steps for the remote URL setup are as follows:
```

Comment:
```suggestion
The usage of a [remote URL as module source](https://developer.hashicorp.com/terraform/language/modules/sources#fetching-archives-over-http) is recommended, because it simplifies the upgrade process. For [local paths as module source](https://developer.hashicorp.com/terraform/language/modules/sources#local-paths), you need to manually overwrite the Terraform files in the Terraform workspace. The steps for the remote URL setup are as follows:
```